tests/integration/test_log_proxy.py

```diff
@@ -57,3 +54,103 @@ async def test_alert_rules_do_fire(ops_test, loki_charm, log_proxy_tester_charm)
         for key in ["juju_application", "juju_model", "juju_model_uuid"]
         for alert in alerts
     )
+    await ops_test.model.applications[tester_app_name].remove()
+    await ops_test.model.block_until(
+        lambda: tester_app_name not in ops_test.model.applications, timeout=300
+    )
+
+
+@pytest.mark.abort_on_fail
+async def test_logproxy_file_logs(ops_test, log_proxy_tester_charm):
+    """Make sure Loki endpoints propagate on scaling."""
+    loki_app_name = "loki"
+    tester_app_name = "log-proxy-tester-file"
+
+    await ops_test.model.deploy(
+        log_proxy_tester_charm,
+        resources=tester_resources,
+        application_name=tester_app_name,
+    )
+    await ops_test.model.block_until(
+        lambda: (
+            len(ops_test.model.applications[loki_app_name].units) > 0
+            and len(ops_test.model.applications[tester_app_name].units) > 0
+        )
+    )
+    await ops_test.model.add_relation(loki_app_name, tester_app_name)
+    await ops_test.model.wait_for_idle(apps=[loki_app_name, tester_app_name], status="active")
+
+    # This is silly, but we need to actually wait to get log data
+    time.sleep(30)
+
+    logs = await loki_api_query(
+        ops_test, loki_app_name, f"juju_application=~'{tester_app_name}',filename=~'.+'"
+    )
+    logger.info("LOGS: {}".format(logs))
+    assert len(logs["values"]) > 0
+
+
+@pytest.mark.abort_on_fail
+async def test_logproxy_syslog_logs(ops_test, log_proxy_tester_charm):
+    """Make sure Loki endpoints propagate on scaling."""
+    loki_app_name = "loki"
+    tester_app_name = "log-proxy-tester-syslog"
+
+    await ops_test.model.deploy(
+        log_proxy_tester_charm,
+        resources=tester_resources,
+        application_name=tester_app_name,
+        config={"syslog": "true", "file_forwarding": "false"},
+    )
+    await ops_test.model.block_until(
+        lambda: (
+            len(ops_test.model.applications[loki_app_name].units) > 0
+            and len(ops_test.model.applications[tester_app_name].units) > 0
+        )
+    )
+    await ops_test.model.add_relation(loki_app_name, tester_app_name)
+    await ops_test.model.wait_for_idle(apps=[loki_app_name, tester_app_name], status="active")
+
+    # This is silly, but we need to actually wait to get log data
+    time.sleep(30)
+
+    logs = await loki_api_query(
+        ops_test, loki_app_name, f"juju_application=~'{tester_app_name}',job=~'.+syslog'"
+    )
+    assert len(logs["values"]) > 0
+
+
+@pytest.mark.abort_on_fail
+async def test_logproxy_logs_to_file_and_syslog(ops_test, log_proxy_tester_charm):
+    """Make sure Loki endpoints propagate on scaling."""
+    loki_app_name = "loki"
+    tester_app_name = "log-proxy-tester-both"
+
+    await ops_test.model.deploy(
```

Comment:
Seems like a `remove` is missing before this line.

Reply:
It is not. Different application names are used every time for this reason.